packages/browser/src/core/task/task-group.ts

```diff
@@ -0,0 +1,29 @@
+export type TaskGroup = {
+  done: () => Promise<void>
+  run: <Operation extends (...args: any[]) => any>(
+    op: Operation
+  ) => ReturnType<Operation>
+}
+
+export const createTaskGroup = (): TaskGroup => {
+  let taskCompletionPromise: Promise<void>
+  let resolvePromise: () => void
+  let count = 0
+
+  return {
+    done: () => taskCompletionPromise,
+    run: (op) => {
+      const returnValue = op()
+
+      if (returnValue instanceof Promise) {
```

Comment:
@chrisradek alerted me to the possible gotchas of “instanceof Promise” (does not work in all environments)  — there should be an “isPromise” or “isThenable” helper in utils that we used instead.

Reply:
Fixed up